search/saddleback_search.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file saddleback_search.cpp
+ * @brief Implementation of saddleback algorithm for 2D arrays.
+ *
+ * @details
+ * Saddleback Algorithm is an algorithm that searches 2D array in linear time,
+ * i.e, O(m + n), where m is row and n is column of 2D array. Also, each row and
+ * column of the matrix should be sorted beforehand for this algorithm to work.
+ *
+ * @author [Hashir Niazi](https://github.com/HashirGJ8842)
+ */
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+/** \namespace search
+ * \brief Algorithms for searching
+ */
+namespace search {
+/** \namespace saddleback
+ * \brief Function for implementing Saddleback Algorithm.
+ */
+namespace saddleback {
+/**
+ * This function implements saddleback
+ * @param matrix , a 2D matrix which is sorted on the basis of rows and columns
+ * @param element , the element which is needed to be searched
+ * @return An std::pair of with row and column populated within it, if the
+ * element is present.
+ * @return An std::pair with (-1, -1), if the element is not present.
+ */
+std::pair<size_t, size_t> saddleback(std::vector<std::vector<int>> matrix,
```

Comment:
Consider using `uint64_t` for non-negative values or `int64_t` for negative values (or their appropriate size). Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)).